packages/shared/src/contexts/BootProvider.tsx

```diff
@@ -112,20 +113,20 @@ export const BootDataProvider = ({
     updatedBootData: Partial<BootCacheData>,
     update = true,
   ) => {
+    const cachedData = JSON.parse(storage.getItem(BOOT_LOCAL_KEY));
     let updatedData = { ...updatedBootData };
     if (update) {
       if (lastAppliedChange) {
         updatedData = { ...lastAppliedChange, ...updatedData };
       }
       setLastAppliedChange(updatedData);
     } else {
-      if (lastAppliedChange) {
+      if (cachedData?.last_modifier !== 'companion' && lastAppliedChange) {
         updatedData = { ...updatedData, ...lastAppliedChange };
       }
       setLastAppliedChange(null);
     }
-
-    const updated = updateLocalBootData(cachedBootData, updatedData);
+    const updated = updateLocalBootData(cachedData, updatedData);
```

Comment:
Not sure I understand the reason for this change

Reply:
Thanks for raising this.
We introduced the lastAppliedChange check which could favorite local data over remote data.
however with the introduction of the companion there is a "third" race condition.

When the companion runs a update we always want to favorite that, (it's the only way this last_modifier get's set to companion)

Else we got into edge cases where you turn of the companion (from the companion)
And the extension would re-enable it again from cache or remote data.

Let me know if that makes sense, happy to showcase it if that helps

Reply:
We need to document it somehow. It will be very hard to maintain this piece without understanding the side effects

Reply:
Let me create a DR for this and describe the flow we choose